### PR TITLE
Fix Key Vault project files

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
We probably just need to update some recorded tests in Azure.Identity.
 ServiceVersion in CryptographyClientOptions.cs https://github.com/Azure/azure-sdk-for-net/blob/cddefe7d93026be8b4ad835d0cb7c64969c4ff46/sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClientOptions.cs#L19 has moved ahead of what is present in our recorded tests.
https://github.com/Azure/azure-sdk-for-net/blob/b015ef33bf4945661d74c4f18411e986558fbefe/sdk/identity/Azure.Identity/tests/SessionRecords/ManagedIdentityCredentialImdsLiveTests/ValidateImdsUserAssignedIdentity.json#L4

This should unblock for now although I don't think reverting the KeyVault package versions make a difference

**Comment by heaths** [comment]
Then all the tests using PackageReferences would break, which are 4 platforms. I think the long term solution needs more thought to how this could work when service versions rev.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml]: We should definitely not comment this out. As we chatted in IM we should update the test projects.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj
Comment (weshaggard) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj]: Any idea how this bad merge happened? Was it incorrectly merged? I would have expected a merge conflict so we would have needed to take both in order to get into this state.

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj]: Not sure how it got merged that way, but the effect would've been the same since the next `Version` would've overwrote this first one. Could have something to do with .gitattributes using `merge=union` for *.csproj. In most other repos, .csproj are treated as any other text file so a merge conflict would've happened (if not resolved automatically). Can't say I've seen this before without `merge=union`.

**Review by weshaggard — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj
Comment (weshaggard) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj]: Thanks for the pointer I think you are correct that is what caused this. After some reading of what union does I don't think we want that for our repo so I'm going to remove it.

Comment (weshaggard) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj]: https://github.com/Azure/azure-sdk-for-net/pull/10440

**Pull request merged**
